### DISCUSSION TITLE
PW dashboard parallel to openhab on raspberryPie

### DISCUSSION THREAD

**Discussion by renekar** [jasonacox/Powerwall-Dashboard#334] [General]
I‘m looking for the opportunity to run the powerwall dashboard on a raspberrypie with a parallel instance of openhab. Any experience of this possible?

**Comment by jasonacox** [comment]
Hi @renekar !

There is a small chance that the default install will work if openhab has grafana and influxdb set up on different port.  You can test by doing this on the raspberypi (you will need to ssh in or use a terminal window):

```bash
# pull the dashboard from github
git clone https://github.com/jasonacox/Powerwall-Dashboard.git
cd Powerwall-Dashboard

# BEFORE you run setup, run verify to see if there are conflicting ports
./verify.sh
```

The Verify tool should be full of errors, but look for the "Service (port)" lines under Grafana and InfluxDB:

<img width="573" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/76d9e1d6-e779-47e9-8912-6261a1dc8584">

If  "GOOD" doesn't show up on either of those, that means you will not have a port conflict and you can proceed with running `setup.sh`.

However, if "GOOD" shows up on either of those, you You will need to modify the dashboard setup [powerwall.yml](https://github.com/jasonacox/Powerwall-Dashboard/blob/main/powerwall.yml) to change the `published` ports for both.  Here is an example where I changed the `influxdb` section to change the published port from 8086 to 8286 (just added 200):

```yml
        ports:
            - target: 8086
              published: 8286
              mode: host
```

You will then need to complete the setup with the manual method shown here: https://github.com/jasonacox/Powerwall-Dashboard/tree/main#option-2---manual-install

If you attempt this, let us know how it goes.

**Comment by renekar** [comment]
Hi, it's verifying at all...

```
Verify Powerwall-Dashboard - Version 2.9.10 - Timezone: America/Los_Angeles
----------------------------------------------------------------------------
This script will attempt to verify all the services needed to run
Powerwall-Dashboard. Use this output when you open an issue for help:
https://github.com/jasonacox/Powerwall-Dashboard/issues/new

 - WARNING: You are missing compose.env

Checking pypowerwall
----------------------------------------------------------------------------
 - Config File pypowerwall.env: ERROR: Missing$
 - Container (pypowerwall): openhabian@openhabian:~/Powerwall-Dashboard $ 
```

**Reply by jasonacox** [reply]
It looks like it crashed.  I'll have to look at the verify.sh error handling. :)

Try this instead - it will show all the TCP ports that are listening on the host.  
```bash
ss -tuln | grep "LISTEN"
```

If you don't see see 9000, 8675, 8676, or 8086, you should be able to run `setup.sh` without a conflict.

**Comment by renekar** [comment]
only 8086, but modified powerwall.yml as mentioned in your first post. 
was able to run the instal script. I have now two instances of Grafana - ip:3000 (openhab) and ip:9000 (power wall dashboard). Any chance to merge?

**Reply by jasonacox** [reply]
Great job!  Feel free to take on the challenge to merge them.  Grafana is probably the easy one as you could manually set up the data sources and then import dashboard.json into the OpenHab version.
